gems/pending/spec/disk/modules/azure_blob_disk_spec.rb

```diff
@@ -0,0 +1,152 @@
+require 'ostruct'
+require 'azure-armrest'
+require 'disk/modules/AzureBlobDisk'
+require 'vcr'
+
+describe AzureBlobDisk do
+  before(:all) do
+    @test_env = TestEnvHelper.new(__FILE__)
+    @test_env.vcr_filter
+
+    @client_id     = @test_env[:azure_client_id]
+    @client_key    = @test_env[:azure_client_key]
+    @tenant_id     = @test_env[:azure_tenant_id]
+    @blob_uri      = @test_env[:blob_uri]
+
+    @test_env.ensure_recording_dir_exists
+  end
+
+  before(:each) do |example|
+    Azure::Armrest::ArmrestService.clear_caches
+    example_id = "#{example.example_group.description}-#{example.metadata[:ex_tag]}"
+    cassette_name = @test_env.cassette_for(example_id)
+    VCR.insert_cassette(cassette_name, :decode_compressed_response => true)
+
+    @azure_config = Azure::Armrest::ArmrestService.configure(
+      :client_id  => @client_id,
+      :client_key => @client_key,
+      :tenant_id  => @tenant_id
+    )
+
+    @storage_account_service = Azure::Armrest::StorageAccountService.new(@azure_config)
+  end
+
+  after(:each) do
+    VCR.eject_cassette
+  end
+
+  describe ".new" do
+    it "should raise an error, given a bad blob uri", :ex_tag => 1 do
```

Comment:
I'm sorry, but I disagree. Variables and metadata should be named based on what they are, not how they're used. What if this were used in multiple ways, would the name keep getting longer?

The script is the consumer of the metadata, not `cassette_for`. The script uses the metadata value to build the string and `cassette_for` is the consumer of the string. Given that this metadata item is only unique within it's group, it shouldn't be used to identify examples to run.

I would say that there should be a comment above the line that constructs the cassette name, and the metadata name should be left alone. It would be much easier to describe it in a comment, than try to accomplish the same thing in a tag name.